lib/kubeclient/common.rb

```diff
@@ -42,14 +48,15 @@ module ClientMixin
     attr_reader :headers
     attr_reader :discovered
 
-    def initialize_client(
+    def initialize_client( # rubocop:disable Metrics/ParameterLists
```

Comment:
see rubocop's PR description.
I believe 8 positional `f(a, b, c, d, e, f, g, h)` would be insane, 
but `f(a, b, c, d, options)` is not so bad, and `f(a, b, c, d, x:, y:, z:, w:)` is strictly cleaner than unpacking `options` inside the function, shouldn't be penalized.

AFAICT rubocop can't make this distinction, but this function is only place >5 total.